### ISSUE TITLE
Ctrl+hover/Cmd+hover does not work as expected

### ISSUE THREAD

**Issue by maxandersen** [GoClipse/goclipse#216]
I'm used to language editors in eclipse that have F3 (go to declaration) to have Ctrl+hover mouse highlight things that are navigatable.

Instead in goeclipse it is kinda of a "lottery" to know if something is possible to open.

If one press F3 it jumps/go to the declaration if it can and otherwise a dialog is shown saying it is not possible and I have to close it and try again.

Is there a reason why eclipse default "highlight declarations" are not working ? bug or expected ?

**Comment by sanmai** [comment]
Possibly related: #111 #70

**Comment by bruno-medeiros** [comment]
@maxandersen Yeah, not a bug, it's a limitation arising from the fact that Goclipse uses an external tool for such semantic operations (as opposed to JDT or CDT that have their semantic engine built-in). So you only know the result of the editor hyperlink (Ctrl + hover) after invoking it. 
That said there are some improvements that could be made to this:

1. Filter Go language keywords from hyperlink words. Fairly trivial.
1. Filter declarations names from hyperlink words, using the source Structure information (ie, the Outline information). Not so trivial.

**Comment by maxandersen** [comment]
thanks for the explanation @bruno-medeiros and I was guessing this was the case.

May I suggest to at least don't show an error dialog when F3 does not work but simply report status warning/error so eclipse will just show in status bar nothign was found for selection rather than having a modal dialog that needs to get dismissed by user ?

**Comment by bruno-medeiros** [comment]
Yeah, could do that, it would be similar to completion tool failures actually.
